File: client/java/src/main/java/io/be/mesh/system/MeshCluster.java
/*
 * Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
 * TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 *
 *
 */
package io.be.mesh.system;

import io.be.mesh.macro.SPI;
import io.be.mesh.mpc.ServiceProxy;
import io.be.mesh.prsim.Cluster;

/**
 * @author coyzeng@gmail.com
 */
@SPI("mesh")
public class MeshCluster implements Cluster {

    private final Cluster cluster = ServiceProxy.proxy(Cluster.class);

    @Override
    public byte[] election(byte[] buff) {
        return cluster.election(buff);
    }

    @Override
    public boolean isLeader() {
        return cluster.isLeader();
    }
}

File: client/python/mesh/system/mesh_cluster.py
#
# Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
# TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
#
#
from mesh.macro import spi
from mesh.mpc import ServiceProxy
from mesh.prsim import Cluster


@spi("mesh")
class MeshCluster(Cluster):

    def __init__(self):
        self.proxy = ServiceProxy.default_proxy(Cluster)

    def election(self, buff: bytes) -> bytes:
        return self.proxy.election(buff)

    def is_leader(self) -> bool:
        return self.proxy.is_leader()
